ManagementPacks/Community.DataOnDemand/Scripts/Watch-TcpConnections.ps1

```diff
@@ -0,0 +1,539 @@
+            [MarshalAs(UnmanagedType.ByValArray, SizeConst = 4)]
+            public byte[] localPort;
+            [MarshalAs(UnmanagedType.ByValArray, SizeConst = 16)]
+            public byte[] remoteAddr;
+            public uint remoteScopeId;
+            [MarshalAs(UnmanagedType.ByValArray, SizeConst = 4)]
+            public byte[] remotePort;
+            public uint state;
+            public uint owningPid;
+        }
+
+        // http://msdn.microsoft.com/en-us/library/windows/desktop/aa366905
+        [StructLayout(LayoutKind.Sequential)]
+        private struct MIB_TCP6TABLE_OWNER_PID
+        {
+            public uint dwNumEntries;
+            [MarshalAs(UnmanagedType.ByValArray, ArraySubType = UnmanagedType.Struct, SizeConst = 1)]
+            public MIB_TCP6ROW_OWNER_PID[] table;
+        }
+        #endregion
+
+        #region enums
+        // http://msdn2.microsoft.com/en-us/library/aa366386.aspx
+        private enum TCP_TABLE_CLASS
+        {
+            TCP_TABLE_BASIC_LISTENER,
+            TCP_TABLE_BASIC_CONNECTIONS,
+            TCP_TABLE_BASIC_ALL,
+            TCP_TABLE_OWNER_PID_LISTENER,
+            TCP_TABLE_OWNER_PID_CONNECTIONS,
+            TCP_TABLE_OWNER_PID_ALL,
+            TCP_TABLE_OWNER_MODULE_LISTENER,
+            TCP_TABLE_OWNER_MODULE_CONNECTIONS,
+            TCP_TABLE_OWNER_MODULE_ALL
+        }
+
+        // http://msdn.microsoft.com/en-us/library/aa366896.aspx
+        private enum MIB_TCP_STATE
+        {
+            MIB_TCP_STATE_CLOSED,
+            MIB_TCP_STATE_LISTEN,
+            MIB_TCP_STATE_SYN_SENT,
+            MIB_TCP_STATE_SYN_RCVD,
+            MIB_TCP_STATE_ESTAB,
+            MIB_TCP_STATE_FIN_WAIT1,
+            MIB_TCP_STATE_FIN_WAIT2,
+            MIB_TCP_STATE_CLOSE_WAIT,
+            MIB_TCP_STATE_CLOSING,
+            MIB_TCP_STATE_LAST_ACK,
+            MIB_TCP_STATE_TIME_WAIT,
+            MIB_TCP_STATE_DELETE_TCB
+        }
+        #endregion
+
+        #region APIs
+        // http://msdn2.microsoft.com/en-us/library/aa366073.aspx
+        private static class IPHelperAPI
+        {
+            [DllImport("iphlpapi.dll", SetLastError = true)]
+            internal static extern uint GetExtendedTcpTable(
+                IntPtr tcpTable,
+                ref int tcpTableLength,
+                bool sort,
+                int ipVersion,
+                TCP_TABLE_CLASS tcpTableType,
+                int reserved);
+        }
+        #endregion
+
+        #region wrapper class
+        private class IPHelperWrapper : IDisposable
+        {
+
+            public const int AF_INET = 2;    // IP_v4 = System.Net.Sockets.AddressFamily.InterNetwork
+            public const int AF_INET6 = 23;  // IP_v6 = System.Net.Sockets.AddressFamily.InterNetworkV6
+
+            // Creates a new wrapper for the local machine
+            public IPHelperWrapper() { }
+
+            // Disposes of this wrapper
+            public void Dispose() { GC.SuppressFinalize(this); }
```

Comment:
The finalizer calls dispose, and this dispose is just making away for an overridden finazlier so... is there any reason we're disposable (Or are we forced by some external code to implement disposable?)